packages/apps/dashboard/ui-2024/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,199 @@
+import { FC, useEffect, useState } from 'react';
+import clsx from 'clsx';
+import { useNavigate } from 'react-router-dom';
+import IconButton from '@mui/material/IconButton';
+import SearchIcon from '@mui/icons-material/Search';
+import CloseIcon from '@mui/icons-material/Close';
+import {
+  InputAdornment,
+  TextField,
+  Select as MuiSelect,
+  SelectChangeEvent,
+  Grid,
+  MenuItem,
+  Box,
+} from '@mui/material';
+import { colorPalette } from '@assets/styles/color-palette';
+import { getNetwork, networks } from '@utils/config/networks';
+import { useBreakPoints } from '@utils/hooks/use-is-mobile';
+import { NetworkIcon } from '@components/NetworkIcon';
+import { useWalletSearch } from '@utils/hooks/use-wallet-search';
+import {
+  endAdornmentInputAdornmentSx,
+  startAdornmentInputAdornmentSx,
+  muiSelectSx,
+  menuItemSx,
+  muiTextFieldInputPropsSx,
+  muiTextFieldSx,
+  gridSx,
+} from './SearchBar.styles';
+
+interface SearchBarProps {
+  className?: string;
+  displaySearchBar?: boolean;
+  isTopBar?: boolean;
+  borderColor?: string;
+  initialInputValue?: string;
+}
+
+const SearchBar: FC<SearchBarProps> = ({
+  className = '',
+  displaySearchBar,
+  borderColor = colorPalette.skyOpacity,
+  initialInputValue = '',
+  isTopBar = false,
+}) => {
+  const { mobile } = useBreakPoints();
+  const [inputValue, setInputValue] = useState<string>(initialInputValue);
+  const [selectValue, setSelectValue] = useState<string>('');
+  const [focus, setFocus] = useState<boolean>(false);
+  const { filterParams, setAddress } = useWalletSearch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const networkName = getNetwork(filterParams.chainId || -1)?.name || '';
+    if (networkName) {
+      setSelectValue(networkName);
+    }
+  }, [filterParams.chainId]);
+
+  useEffect(() => {
+    setInputValue(filterParams.address);
+  }, [filterParams.address]);
+
+  useEffect(() => {
+    if (initialInputValue) {
+      setInputValue(initialInputValue);
+    }
+  }, [initialInputValue]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const handleSelectChange = (event: SelectChangeEvent<string>) => {
+    const chainId = Number(event.target.value);
+    const networkName = getNetwork(chainId)?.name || '';
```

Comment:
As function itself - yes, it can. But for `Select` component we provide a specific list of `networks` that is already filtered by `getNetwork` (to have chain name and id). So IMU `getNetwork` will return `undefined` only if `chainId` selected is empty, so I though maybe we could improve something there as well. But taking into account that we have different `renderValue` for empty state - it doesn't makes sense to touch it